nix/nim-status/default.nix

```diff
@@ -0,0 +1,85 @@
+{ newScope
+, writeScript
+, xcodeWrapper
+, pkgs
+, stdenv
+, lib }:
+let
+  callPackage = newScope {};
+
+in rec {
+  sqlcipher = callPackage ./sqlcipher.nix { platform = "android"; arch = "386"; };
+  nim-sqlcipher = callPackage ./nim-sqlcipher.nix { platform = "android"; arch = "386"; };
+
+  android-x86 = callPackage ./nim-status.nix { 
+    platform = "android"; 
+    arch = "386"; 
+  };
+  android-arm = callPackage ./nim-status.nix { 
+    platform = "androideabi"; 
+    arch = "arm";
+  };
+  android-arm64 = callPackage ./nim-status.nix {
+    platform = "android"; 
+    arch = "arm64";
+  };
+
+  ios-x86 = callPackage ./nim-status.nix { 
+    platform = "ios"; 
+    arch = "386"; 
+  };
+  ios-arm = callPackage ./nim-status.nix {
+    platform = "ios"; 
+    arch = "arm"; 
+  };
+  ios-arm64 = callPackage ./nim-status.nix {
+    platform = "ios"; 
+    arch = "arm64"; 
+  };
+
+  android = stdenv.mkDerivation {
+    name = "nim-status-android-builder";
+    buildInputs = [ pkgs.coreutils ];
+    builder = writeScript "nim-status-android-builder.sh"
```

Comment:
Actually, this whole derivation could be done with [`buildEnv`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/buildenv/default.nix) or [`symlinkJoin`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/buildenv/default.nix). Something like:
```nix
{
  android-test = pkgs.symlinkJoin {
    name = "nim-status-android";
    paths = [ android-x86.arch android-arm.arch android-arm64.arch ];
  };
}
```
This would just require the `nim-status.nix` derivation to add `arch` to `outputs` and symlink the right folder:
```nix
{
  outputs = ["out" "arch"];
  installPhase = ''
    set -x
    mkdir -p $out $arch
    cp nimcache/nim_status_go/shim.h $out/nim_status.h
    cp ./nim_status/c/go/include/nimbase.h $out/
    mv nim_status_go.a $out/libnim_status.a
    ln -s $out $arch/x86
  '';
}
```
Of course the `x86` part would have to be dynamic. Or, if we only ever use them with the sub-folder then we don't even need the `arch` output. Just make the default `out` put stuff in the subfolder from the start.